Sources/Extensions/StringExtension.swift

```diff
@@ -46,6 +46,36 @@ extension String {
         return NSPredicate(format: "SELF MATCHES %@", emailRegex).evaluate(with: self)
     }
 
+    public var isIP4Address: Bool {
+        return confirmIP4isValid(ip4: self)
+    }
+    
+    public var isIP6Address: Bool {
+        return confirmIP6isValid(ip6: self)
+    }
+    
+    public var isIPAddress: Bool {
+        return confirmIP4isValid(ip4: self) || confirmIP6isValid(ip6: self)
+    }
+
+    private func confirmIP4isValid(ip4: String) -> Bool {
+        var valid = false
+        var sin = sockaddr_in()
+        if ip4.withCString({ cstring in inet_pton(AF_INET, cstring, &sin.sin_addr)}) == 1 {
+            valid = true
+        }
+        return valid
```

Comment:
Why not returning directly `ip4.withCString({ cstring in inet_pton(AF_INET, cstring, &sin.sin_addr)}) == 1` instead of having a `valid` variable?

Reply:
I had it written in a different context previously where what I did made sense and lazily copy/pasted. :P 

I'll fix it.